lib/adal-angular.js

```diff
@@ -198,6 +199,16 @@
                 }
 
                 var routeChangeHandler = function (e, nextRoute) {
+                    if (!_paths) {
+                        _paths = $injector.get('$route').routes;
+                        for (var route in _paths) {
+                            if (_paths[route].hasOwnProperty('templateUrl') && !isADLoginRequired(_paths[route], _adal.config)) {
+                                if (!(_adal.config.anonymousEndpoints.indexOf(_paths[route].templateUrl) > -1)) {
```

Comment:
use isAnonymousEndpoint(url) method here. Its possible that user added "App/Views" to anonymous endpoints in which case we don't need to add "App/Views/Home.html" as every route under "App/Views" is considered anonymous. Your check does a full string match.